File: UiComponents/src/main/java/com/waycool/uicomponents/utils/AppUtil.java
package com.waycool.uicomponents.utils;

import android.content.Context;
import android.os.Handler;
import android.provider.Settings;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.Toast;

import com.bumptech.glide.Glide;
import com.squareup.picasso.Picasso;
import com.waycool.uicomponents.R;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;


public class AppUtil {

    public static void showUrlImage(Context context, String imageUrl, ImageView image) {
        Glide.with(context)
                .load("https://outgrow.blob.core.windows.net/outgrowstorage/Prod/CropLogo/Apple.svg")
                .fitCenter()
                .placeholder(R.drawable.ic_outgrow)
                .error(R.drawable.ic_outgrow)
                .into(image);
    }

    public static void showBannerImage(Context context, String imageUrl, ImageView image) {
        Picasso.get()
                .load(imageUrl)
                .centerCrop()
                .placeholder(R.drawable.outgrow_logo_new)
                .into(image);
    }
    public static void showImageDrawable(Context context, int imageUrl, ImageView image) {
        Picasso.get().load(imageUrl).into(image);
    }
    public static void showFitPicasoo(Context context, String imageUrl, ImageView image) {
        Picasso.get()
                .load(imageUrl)
                .fit()
                .placeholder(R.drawable.outgrow_logo_new)
                .into(image);
    }

    public static void showLoadImage(Context context, ImageView image) {
        try {
            Glide.with(context).asGif().load(R.raw.loading).into(image);
        }catch(Exception e){
            e.printStackTrace();
        }
    }
    public static void Toast(Context context,String msg){
        try {
            Toast.makeText(context,msg, Toast.LENGTH_SHORT).show();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
    public static void setProgressVisible(Context context, View loadingScreen, ImageView loadingImage ){
        try{
            loadingScreen.setVisibility(View.VISIBLE);
            AppUtil.showLoadImage(context,loadingImage);
        }catch(Exception e){
            e.printStackTrace();
        }
    }
    public static void setProgressGone(Context context,View loadingScreen){
        try {
            loadingScreen.setVisibility(View.GONE);
        }catch(Exception e){
            e.printStackTrace();
        }
    }
    public static String getAndroidId(Context context) {
        String androidId = Settings.Secure.getString(context.getContentResolver(),
                Settings.Secure.ANDROID_ID);
        return androidId;
    }
    public static String changeDateFormat(String givenDate) {
        String convertedDate = "";
        String date = givenDate;
        SimpleDateFormat input = new SimpleDateFormat("yyyy-MM-dd");
        SimpleDateFormat output = new SimpleDateFormat("dd MMM yy");
        try {
            Date inputDate = input.parse(date);                 // parse input
            convertedDate = output.format(inputDate); //get output
        } catch (ParseException e) {
            e.printStackTrace();
        }
        Log.d("dateCon", convertedDate);
        return convertedDate;
    }

    public static void handlerSet(Handler handler, Runnable runnable, int delayTime){
        handler.removeCallbacks(runnable);
        handler.postDelayed(runnable,delayTime);
    }


}

File: UiComponents/src/main/java/com/waycool/uicomponents/utils/AppUtil.kt
package com.waycool.uicomponents.utils

import android.content.Context
import android.os.Handler
import android.provider.Settings
import android.util.Log
import android.view.View
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.squareup.picasso.Picasso
import com.waycool.uicomponents.R
import java.text.ParseException
import java.text.SimpleDateFormat

object AppUtil {
    fun showUrlImage(context: Context?, imageUrl: String?, image: ImageView?) {
        Glide.with(context!!)
            .load("https://outgrow.blob.core.windows.net/outgrowstorage/Prod/CropLogo/Apple.svg")
            .fitCenter()
            .placeholder(R.drawable.ic_outgrow)
            .error(R.drawable.ic_outgrow)
            .into(image!!)
    }

    fun showBannerImage(context: Context?, imageUrl: String?, image: ImageView?) {
        Picasso.get()
            .load(imageUrl)
            .centerCrop()
            .placeholder(R.drawable.outgrow_logo_new)
            .into(image)
    }

    fun showImageDrawable(context: Context?, imageUrl: Int, image: ImageView?) {
        Picasso.get().load(imageUrl).into(image)
    }

    fun showFitPicasoo(context: Context?, imageUrl: String?, image: ImageView?) {
        Picasso.get()
            .load(imageUrl)
            .fit()
            .placeholder(R.drawable.outgrow_logo_new)
            .into(image)
    }

    fun showLoadImage(context: Context?, image: ImageView?) {
        try {
            Glide.with(context!!).asGif().load(R.raw.loading).into(image!!)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun Toast(context: Context?, msg: String?) {
        try {
            android.widget.Toast.makeText(context, msg, android.widget.Toast.LENGTH_SHORT).show()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun setProgressVisible(context: Context?, loadingScreen: View, loadingImage: ImageView?) {
        try {
            loadingScreen.visibility = View.VISIBLE
            showLoadImage(context, loadingImage)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun setProgressGone(context: Context?, loadingScreen: View) {
        try {
            loadingScreen.visibility = View.GONE
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun getAndroidId(context: Context): String {
        return Settings.Secure.getString(
            context.contentResolver,
            Settings.Secure.ANDROID_ID
        )
    }

    fun changeDateFormat(givenDate: String): String {
        var convertedDate = ""
        val input = SimpleDateFormat("yyyy-MM-dd")
        val output = SimpleDateFormat("dd MMM yy")
        try {
            val inputDate = input.parse(givenDate) // parse input
            convertedDate = output.format(inputDate) //get output
        } catch (e: ParseException) {
            e.printStackTrace()
        }
        Log.d("dateCon", convertedDate)
        return convertedDate
    }

    fun handlerSet(handler: Handler, runnable: Runnable?, delayTime: Int) {
        handler.removeCallbacks(runnable!!)
        handler.postDelayed(runnable, delayTime.toLong())
    }
}